clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5078,6 +5078,13 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                                  options::OPT_fno_trigraphs))
       if (A != Std)
         A->render(Args, CmdArgs);
+  } else if (IsSYCL && types::isCXX(InputType)) {
+    // For DPC++, we default to -std=c++17 for all compilations.  Use of -std
+    // on the command line will override.
+    CmdArgs.push_back("-std=c++17");
```

Comment:
Just curious. I can't see how use of `-std` will override the DPC++ setting as I don't see any check for user-provided `-std` here.

Reply:
The user set -std is checked with the `if` conditional above on line 5066.